src/context/authContext.js

```diff
@@ -0,0 +1,44 @@
+import { createContext, useContext, useReducer } from "react"
+
+const AuthContext = createContext()
+
+const AuthProvider=({children}) => {
+    const initialSignInStatus = {
+      status: localStorage.getItem("token")===null?false:true,
+      encodedToken: localStorage.getItem("token"),
+    };
+
+
+    const signInReducer = (signInStatus,action) => {
+
+        switch(action.type){
+            case "SET_USER":
```

Comment:
you can create constants for these actions and import and use them everywhere this is a good practice and will remove alot of typos.

Reply:
Will do this in refactoring